### ISSUE TITLE
Wrap modules within derivations

### ISSUE THREAD

**Issue by gytis-ivaskevicius** [nix-community/home-manager#2267]
TL;DR: I'd like to see something along the lines of this:
```nix
let
  dunst = lib.wrapDunst pkgs.dunst {
	settings = {
	    global = {
	      geometry = "300x5-30+50";
	      transparency = 10;
	      frame_color = "#eceff1";
	      font = "Droid Sans 9";
	    };
	  
	    urgency_normal = {
	      background = "#37474f";
	      foreground = "#eceff1";
	      timeout = 10;
	    };
	  };
  }; 
in 
{...}
```
Why? 
- Derivations are super nice self contained units which you can easily share and show others. This way generic `nix run` would be enough for other people try other people's configs.
- Even tho .config is meant for such stuff - I don't think that there is a need for Nix to polute it.

How should external configurations like services be handled?
For more complex stuff it should be left as it is. For packages like dunst wrapper should expose `services` attribute or something along those lines. (Tho I'd be fine with just leaving everything as it is)

Also maybe it would make sense to move such simple config builders to nixpkgs?

Anyways, I would like to hear your guys thoughts and Ill implement some POC as soon as one of the maintainers gives a green flag.

**Comment by ambroisie** [comment]
I'm personally against using wrapped executables, `XDG_CONFIG_HOME` is supposed to be used for precisely this reason, and using home-manager to do that just like I use NixOS to manage my `/etc` feels natural to me.

**Comment by berbiche** [comment]
I agree that one of the issue of HM is its inability to configure a given program multiple times.
It's probably possible to work around this using nixpkgs' `evalConfig` and some scripting.

IMO your project should be standalone.
